src/object/tcb.c

```diff
@@ -162,24 +184,15 @@ void tcbSchedDequeue(tcb_t *tcb)
         idx = ready_queues_index(dom, prio);
         queue = NODE_STATE_ON_CORE(ksReadyQueues[idx], tcb->tcbAffinity);
 
-        if (tcb->tcbSchedPrev) {
-            tcb->tcbSchedPrev->tcbSchedNext = tcb->tcbSchedNext;
-        } else {
-            queue.head = tcb->tcbSchedNext;
-            if (likely(!tcb->tcbSchedNext)) {
-                removeFromBitmap(SMP_TERNARY(tcb->tcbAffinity, 0), dom, prio);
-            }
-        }
+        new_queue = tcb_queue_remove(queue, tcb);
```

Comment:
If `queue` is consumed by `tcb_queue_remove` and then can't the new result be assigned back to the original queue variable instead of using an additional new_queue variable? This seems that it'd be more inline with the original behavior too?

Reply:
It could, and I assume the compiler will actually do that. It's nicer for verification to have a new name, so that `queue` is not assigned to in the method and can be lifted out as a variable that doesn't depend on the state any more.